DragAndDrop/src/main/java/com/microsoft/device/display/samples/draganddrop/fragment/DragSourceFragment.kt

```diff
@@ -34,42 +43,64 @@ class DragSourceFragment : Fragment(), View.OnLongClickListener {
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
+    ): View {
         binding = DragAndDropSourceLayoutBinding.inflate(inflater, container, false)
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        bindUI(view)
+        prepareDragListeners()
     }
 
     /**
      * Binds all UI components
      */
-    private fun bindUI(view: View) {
-        val dragTextView = view.findViewById<TextView>(R.id.drag_text_view)
-        val dragImageView = view.findViewById<ImageView>(R.id.drag_image_view)
-        dragTextView.tag = "text_view"
-        dragImageView.tag = "image_view"
-        dragTextView.setOnLongClickListener(this)
-        dragImageView.setOnLongClickListener(this)
-    }
+    private fun prepareDragListeners() {
+        DragStartHelper(binding.dragTextView) { targetView, _ ->
+            val text = (targetView as TextView).text
+            val dragClipData = ClipData.newPlainText(ClipDescription.MIMETYPE_TEXT_PLAIN, text)
+            val dragShadowBuilder = View.DragShadowBuilder(targetView)
+            targetView.startDragAndDrop(dragClipData, dragShadowBuilder, null, DRAG_FLAG_GLOBAL)
+        }.attach()
 
-    override fun onLongClick(view: View): Boolean {
-        startDragAndDropForView(view)
-        return true
-    }
+        DragStartHelper(binding.dragImageView) { view, _ ->
+            val folder = File(requireContext().filesDir, DRAG_FOLDER_NAME)
+            if (!folder.exists()) {
+                folder.mkdir()
+            }
+            val imageFile = File(folder, DRAG_FILE_NAME)
 
-    /**
-     * Starts the drag and drop action for the given [View]
-     *
-     * @param view The drag source
-     */
-    private fun startDragAndDropForView(view: View) {
-        val clipDataItem = ClipData.Item(view.tag.toString())
-        val mimeTypes = arrayOf(view.mimeType?.value)
-        val clipData = ClipData(view.tag.toString(), mimeTypes, clipDataItem)
-        view.startDragAndDrop(clipData, View.DragShadowBuilder(view), view, 0)
+            if (!imageFile.exists()) {
+                ByteArrayOutputStream().use { bos ->
```

Comment:
You can create another function for 75-82 lines, something like saveBitmap, saveFile, etc